File: cs/demo/Ice/nested/NestedI.cs
// **********************************************************************
//
// Copyright (c) 2003-2014 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

using Demo;

class NestedI : NestedDisp_
{
    internal NestedI(NestedPrx self)
    {
        _self = self;
    }
    
    public override void nestedCall(int level, NestedPrx proxy, Ice.Current current)
    {
        System.Console.Out.WriteLine("" + level);
        if(--level > 0)
        {
            proxy.nestedCall(level, _self);
        }
    }
    
    private NestedPrx _self;
}

File: cpp/demo/Ice/nested/NestedI.cpp
// **********************************************************************
//
// Copyright (c) 2003-2013 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

#include <Ice/Ice.h>
#include <NestedI.h>

using namespace std;
using namespace Ice;
using namespace Demo;

NestedI::NestedI(const NestedPrx& self) :
    _self(self)
{
}

void
NestedI::nestedCall(Int level, const NestedPrx& proxy, const Ice::Current&)
{
    cout << level << endl;
    if(--level > 0)
    {
        proxy->nestedCall(level, _self);
    }
}
